internal/controller/server_controller.go

```diff
@@ -386,7 +387,23 @@ func (r *ServerReconciler) updateServerStatus(ctx context.Context, log logr.Logg
 		return fmt.Errorf("failed to get system info for Server: %w", err)
 	}
 
+	storages, err := bmcClient.GetStorages(server.Spec.UUID)
+	if err != nil {
+		return fmt.Errorf("failed to get storages for Server: %w", err)
+	}
 	serverBase := server.DeepCopy()
+	server.Status.Storages = nil
+	for _, storage := range storages {
+		q := resource.NewQuantity(storage.SizeBytes, resource.BinarySI)
+		server.Status.Storages = append(server.Status.Storages, metalv1alpha1.Storage{
+			Name:     storage.Name,
+			Model:    storage.Model,
+			Capacity: *q,
```

Comment:
Assign the `resource.NewQuantity` directly here after changing the field value to a pointer (see comment above).